Formula/f/fasmg.rb

```diff
@@ -0,0 +1,92 @@
+class Fasmg < Formula
+  desc "New assembly engine designed as a successor of the one used by flat assembler 1"
+  homepage "https://flatassembler.net"
+  url "https://flatassembler.net/fasmg.kp60.zip"
+  version "kp60"
+  sha256 "cd18f546c04007226fe0ce5bea874d14382911ed4f5a1995f6bde913e7e4e751"
+  license "BSD-3-Clause"
+  head "https://github.com/tgrysztar/fasmg.git", branch: "master"
+
+  livecheck do
+    url "https://flatassembler.net/download.php"
+    regex(/href=.*?fasmg\.([0-9a-zA-Z]+)\.zip/i)
+  end
+
+  depends_on arch: :x86_64
```

Comment:
If this doesn't support either macOS or Linux ARM: it's probably not a good fit for Homebrew given most of our users are on macOS ARM.

Reply:
@MikeMcQuaid what is the proper way to define something as x86_64 only and requiring rosetta for arm64?

Reply:
Not sure if we have a DSL for this for formulae, potentially only casks. @homebrew/maintainers any ideas?

Reply:
We don't, but this does not prevent Rosetta usage. It requires a Rosetta install of Homebrew

Reply:
We shouldn't be adding new formulae that need a Rosetta install of Homebrew.